stm/stm.ml

```diff
@@ -1442,7 +1440,6 @@ end = struct (* {{{ *)
   let task_match age t =
     match age, t with
     | Fresh, BuildProof { t_states } ->
-        not !cur_opt.async_proofs_full ||
```

Comment:
This Inverts the normal logic

Reply:
Well spotted! Actually, I had fixed it but forgot to push.

Reply:
I think I'd like a trace about the perspective to stay in the code.
Maybe define a "is_inside_perspective" that ccontains (commented out) the code below and is constant to true.